// @alicloud/dysmsapi20170525/dist/models/GetSmsTemplateListResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSmsTemplateListResponseBody = exports.GetSmsTemplateListResponseBodyData = exports.GetSmsTemplateListResponseBodyDataList = exports.GetSmsTemplateListResponseBodyDataListTemplateTag = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetSmsTemplateListResponseBodyDataListTemplateTag extends $dara.Model {
    static names() {
        return {
            tagKey: 'TagKey',
            tagValue: 'TagValue',
        };
    }
    static types() {
        return {
            tagKey: 'string',
            tagValue: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSmsTemplateListResponseBodyDataListTemplateTag = GetSmsTemplateListResponseBodyDataListTemplateTag;
class GetSmsTemplateListResponseBodyDataList extends $dara.Model {
    static names() {
        return {
            auditStatus: 'AuditStatus',
            gmtCreate: 'GmtCreate',
            signName: 'SignName',
            templateCode: 'TemplateCode',
            templateName: 'TemplateName',
            templateTag: 'TemplateTag',
            templateType: 'TemplateType',
            usableStateList: 'UsableStateList',
            workOrderId: 'WorkOrderId',
        };
    }
    static types() {
        return {
            auditStatus: 'number',
            gmtCreate: 'string',
            signName: 'string',
            templateCode: 'string',
            templateName: 'string',
            templateTag: { 'type': 'array', 'itemType': GetSmsTemplateListResponseBodyDataListTemplateTag },
            templateType: 'number',
            usableStateList: { 'type': 'array', 'itemType': 'string' },
            workOrderId: 'string',
        };
    }
    validate() {
        if (Array.isArray(this.templateTag)) {
            $dara.Model.validateArray(this.templateTag);
        }
        if (Array.isArray(this.usableStateList)) {
            $dara.Model.validateArray(this.usableStateList);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSmsTemplateListResponseBodyDataList = GetSmsTemplateListResponseBodyDataList;
class GetSmsTemplateListResponseBodyData extends $dara.Model {
    static names() {
        return {
            list: 'List',
            pageNo: 'PageNo',
            pageSize: 'PageSize',
            total: 'Total',
        };
    }
    static types() {
        return {
            list: { 'type': 'array', 'itemType': GetSmsTemplateListResponseBodyDataList },
            pageNo: 'number',
            pageSize: 'number',
            total: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.list)) {
            $dara.Model.validateArray(this.list);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSmsTemplateListResponseBodyData = GetSmsTemplateListResponseBodyData;
class GetSmsTemplateListResponseBody extends $dara.Model {
    static names() {
        return {
            accessDeniedDetail: 'AccessDeniedDetail',
            code: 'Code',
            data: 'Data',
            message: 'Message',
            requestId: 'RequestId',
            success: 'Success',
        };
    }
    static types() {
        return {
            accessDeniedDetail: 'string',
            code: 'string',
            data: GetSmsTemplateListResponseBodyData,
            message: 'string',
            requestId: 'string',
            success: 'boolean',
        };
    }
    validate() {
        if (this.data && typeof this.data.validate === 'function') {
            this.data.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSmsTemplateListResponseBody = GetSmsTemplateListResponseBody;
