// @alicloud/dysmsapi20170525/dist/models/GetSmsOcrOssInfoResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSmsOcrOssInfoResponseBody = exports.GetSmsOcrOssInfoResponseBodyModel = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class GetSmsOcrOssInfoResponseBodyModel extends $dara.Model {
    static names() {
        return {
            accessKeyId: 'AccessKeyId',
            bucket: 'Bucket',
            expireTime: 'ExpireTime',
            host: 'Host',
            policy: 'Policy',
            signature: 'Signature',
            startPath: 'StartPath',
        };
    }
    static types() {
        return {
            accessKeyId: 'string',
            bucket: 'string',
            expireTime: 'string',
            host: 'string',
            policy: 'string',
            signature: 'string',
            startPath: 'string',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSmsOcrOssInfoResponseBodyModel = GetSmsOcrOssInfoResponseBodyModel;
class GetSmsOcrOssInfoResponseBody extends $dara.Model {
    static names() {
        return {
            accessDeniedDetail: 'AccessDeniedDetail',
            code: 'Code',
            message: 'Message',
            model: 'Model',
            requestId: 'RequestId',
            success: 'Success',
        };
    }
    static types() {
        return {
            accessDeniedDetail: 'string',
            code: 'string',
            message: 'string',
            model: GetSmsOcrOssInfoResponseBodyModel,
            requestId: 'string',
            success: 'boolean',
        };
    }
    validate() {
        if (this.model && typeof this.model.validate === 'function') {
            this.model.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.GetSmsOcrOssInfoResponseBody = GetSmsOcrOssInfoResponseBody;
