// @alicloud/dypnsapi20170525/dist/models/QueryGateVerifyStatisticPublicResponseBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryGateVerifyStatisticPublicResponseBody = exports.QueryGateVerifyStatisticPublicResponseBodyData = exports.QueryGateVerifyStatisticPublicResponseBodyDataDayStatistic = void 0;
// This file is auto-generated, don't edit it
const $dara = __importStar(require("@darabonba/typescript"));
class QueryGateVerifyStatisticPublicResponseBodyDataDayStatistic extends $dara.Model {
    static names() {
        return {
            statisticDateStr: 'StatisticDateStr',
            totalFail: 'TotalFail',
            totalSuccess: 'TotalSuccess',
            totalUnknown: 'TotalUnknown',
        };
    }
    static types() {
        return {
            statisticDateStr: 'string',
            totalFail: 'number',
            totalSuccess: 'number',
            totalUnknown: 'number',
        };
    }
    validate() {
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryGateVerifyStatisticPublicResponseBodyDataDayStatistic = QueryGateVerifyStatisticPublicResponseBodyDataDayStatistic;
class QueryGateVerifyStatisticPublicResponseBodyData extends $dara.Model {
    static names() {
        return {
            dayStatistic: 'DayStatistic',
            total: 'Total',
            totalFail: 'TotalFail',
            totalSuccess: 'TotalSuccess',
            totalUnknown: 'TotalUnknown',
        };
    }
    static types() {
        return {
            dayStatistic: { 'type': 'array', 'itemType': QueryGateVerifyStatisticPublicResponseBodyDataDayStatistic },
            total: 'number',
            totalFail: 'number',
            totalSuccess: 'number',
            totalUnknown: 'number',
        };
    }
    validate() {
        if (Array.isArray(this.dayStatistic)) {
            $dara.Model.validateArray(this.dayStatistic);
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryGateVerifyStatisticPublicResponseBodyData = QueryGateVerifyStatisticPublicResponseBodyData;
class QueryGateVerifyStatisticPublicResponseBody extends $dara.Model {
    static names() {
        return {
            code: 'Code',
            data: 'Data',
            message: 'Message',
            requestId: 'RequestId',
        };
    }
    static types() {
        return {
            code: 'string',
            data: QueryGateVerifyStatisticPublicResponseBodyData,
            message: 'string',
            requestId: 'string',
        };
    }
    validate() {
        if (this.data && typeof this.data.validate === 'function') {
            this.data.validate();
        }
        super.validate();
    }
    constructor(map) {
        super(map);
    }
}
exports.QueryGateVerifyStatisticPublicResponseBody = QueryGateVerifyStatisticPublicResponseBody;
